awssql/utils/sliding_expiration_cache.go

```diff
@@ -74,6 +74,8 @@ func (c *SlidingExpirationCache[T]) Put(key string, value T, itemExpiration time
 }
 
 func (c *SlidingExpirationCache[T]) Get(key string, itemExpiration time.Duration) (T, bool) {
+	c.cleanupIfExpired(key)
```

Comment:
I also don't think we need the condition in line 83 anymore of `|| shouldCleanup(...)`? Since we would've already cleaned up by this point